backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/util/git/GitUtils.java

```diff
@@ -104,12 +113,85 @@ public static SonarLintBlameResult blameWithFilesGitCommand(Path projectBaseDir,
 
     try {
       var blameResult = blameCommand.call();
+      long endTime = System.currentTimeMillis();
+      LOG.info("blameWithFilesGitCommand took " + (endTime - startTime) + " ms");
       return new SonarLintBlameResult(blameResult, gitRepoRelativeProjectBaseDir);
     } catch (GitAPIException e) {
       throw new IllegalStateException("Failed to blame repository files", e);
     }
   }
 
+  private static boolean isGitAvailable() {
```

Comment:
Only run this once and cache this information (e.g. with a static class attribute). Running this is also costly for every Git command :smile: